akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2Compliance.scala

```diff
@@ -44,9 +44,12 @@ private[http2] object Http2Compliance {
   final def requireZeroStreamId(id: Int): Unit =
     if (id != 0) throw new IllegalHttp2StreamIdException(id, "MUST BE == 0.")
 
+  final def requireNonZeroStreamId(id: Int): Unit =
+    if (id == 0) throw new Http2ProtocolException(ErrorCode.PROTOCOL_ERROR, "Stream ID MUST be > 0") // cause GOAWAY
+
   final def requirePositiveWindowUpdateIncrement(streamId: Int, increment: Int): Unit =
     if (increment <= 0)
-      if (streamId == 0) throw new Http2ProtocolException(ErrorCode.PROTOCOL_ERROR, "WINDOW_UPDATE MUST be > 0, was: " + increment) // cause GOAWAY
+      if (streamId == 0) throw new Http2ProtocolException(ErrorCode.PROTOCOL_ERROR, "Stream ID MUST be > 0") // cause GOAWAY
```

Comment:
Oops, thanks for fixing.

Reply:
Maybe just use `requireNonZeroStreamId` for that?

Reply:
hm actually looking at https://tools.ietf.org/html/rfc7540#section-6.9 a 0 stream id here is fine, removing that check altogether.

Reply:
I did not look closely enough but the tests saved us: a 0 stream id is fine, but a 0 window increment with stream id 0 should be treated as a connection error where a 0 window increment on another stream should be treated as a stream error. Fixing.